DataFormats/TrackerRecHit2D/interface/VectorHit.h

```diff
@@ -0,0 +1,156 @@
+#ifndef TrackerRecHit2D_VectorHit_h
+#define TrackerRecHit2D_VectorHit_h
+
+/** \class VectorHit
+ *
+ * 4-parameter RecHits for Phase2 Tracker (x,y, dx/dz, dy/dz)
+ *
+ * $Date: 2015/03/30 $
+ * \author Erica Brondolin
+ *
+ */
+
+#include "DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/OmniClusterRef.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+
+#include "DataFormats/TrackingRecHit/interface/KfComponentsHolder.h"
+
+#include "TkCloner.h"
+
+class VectorHit GCC11_FINAL : public BaseTrackerRecHit {
+
+  public:
+
+  typedef OmniClusterRef::Phase2Cluster1DRef ClusterRef;
+
+  VectorHit() : thePosition(), theDirection(), theCovMatrix(), theDimension(0) { setType(bad); }
+
+  VectorHit(const VectorHit& vh) ;
+
+  VectorHit(const GeomDet& idet, const LocalPoint& posInner, const LocalVector& dir,
+            const AlgebraicSymMatrix& covMatrix, const double& Chi2,
+            OmniClusterRef const& lower, OmniClusterRef const& upper) ;
+
+  VectorHit(const GeomDet& idet, const VectorHit2D& vh2Dzx, const VectorHit2D& vh2Dzy,
+            OmniClusterRef const& lower, OmniClusterRef const& upper) ;
+
+  ~VectorHit() ;
+
+  virtual VectorHit* clone() const override { return new VectorHit(*this);}
+#ifndef __GCCXML__
+  virtual RecHitPointer cloneSH() const override { return std::make_shared<VectorHit>(*this);}
+#endif
+
+  virtual bool sharesInput( const TrackingRecHit* other, SharedInputType what) const override;
+  bool sharesClusters(VectorHit const & h1, VectorHit const & h2,
+                      SharedInputType what) const ;
+
+  // Parameters of the segment, for the track fit
+  // For a 4D segment: (dx/dz,dy/dz,x,y)
+
+  virtual AlgebraicVector parameters() const override;
+  virtual void getKfComponents( KfComponentsHolder & holder ) const override { getKfComponents4D(holder); }
+  void getKfComponents4D( KfComponentsHolder & holder ) const ;
+
+  // returning methods
+  virtual LocalVector localDirection() const { return theDirection; }
+  AlgebraicSymMatrix parametersError() const override ;
+  virtual LocalError localDirectionError() const ;
+  Global3DVector globalDirection() const;
+
+  virtual double chi2() const { return theChi2; }
+  virtual int dimension() const override { return theDimension; }
+
+  std::pair<double,double> curvatureORphi(std::string curvORphi = "curvature") const ;
```

Comment:
Is this function implemented anywhere (I didn't spot). If not, remove the declaration (mostly I want to get rid of string argument).

Reply:
Answering myself, yes it is (I missed it because GitHub hid it from me).

Can we either split this to two functions `curvature` and `phi` (for cleaner interface), or return all 3 (as a struct) since they are all computed on one go (for better performance)?